[
  {
    "id" : "fced7d78-ee83-4d29-9f16-33ccb9294ff7",
    "prId" : 49844,
    "prUrl" : "https://github.com/flutter/flutter/pull/49844#pullrequestreview-351234827",
    "prSource" : "GitHub",
    "comments" : [
      {
        "id" : "97e8a10f-6d84-4d91-ba10-47ab007e61f4",
        "parentId" : null,
        "authorId" : "e73873df-c6ef-4dd5-ad17-dc7a574dcf0b",
        "body" : "This would output the message for any widget test which incidentally built a network image. It doesn't seem desirable if their test was otherwise working.",
        "createdAt" : "2020-01-30T23:13:06Z",
        "updatedAt" : "2020-01-31T00:55:42Z",
        "lastEditedBy" : "e73873df-c6ef-4dd5-ad17-dc7a574dcf0b",
        "tags" : [
        ]
      },
      {
        "id" : "ab2df88c-e262-4c2d-ab54-47dbacde4e31",
        "parentId" : "97e8a10f-6d84-4d91-ba10-47ab007e61f4",
        "authorId" : "db47316c-0a85-441a-8891-3b84e8bfe474",
        "body" : "Is there some way to just do this on a test failure?",
        "createdAt" : "2020-01-31T00:05:03Z",
        "updatedAt" : "2020-01-31T00:55:42Z",
        "lastEditedBy" : "db47316c-0a85-441a-8891-3b84e8bfe474",
        "tags" : [
        ]
      },
      {
        "id" : "8a90cc31-35d5-467b-bc11-933c8eab5000",
        "parentId" : "97e8a10f-6d84-4d91-ba10-47ab007e61f4",
        "authorId" : "db47316c-0a85-441a-8891-3b84e8bfe474",
        "body" : "I could try to use test_package.printOnFailure, but I'm not sure how to test that anymore - hah",
        "createdAt" : "2020-01-31T00:11:49Z",
        "updatedAt" : "2020-01-31T00:55:42Z",
        "lastEditedBy" : "db47316c-0a85-441a-8891-3b84e8bfe474",
        "tags" : [
        ]
      },
      {
        "id" : "9052527b-16c4-4362-a7c3-ce54337cb9bc",
        "parentId" : "97e8a10f-6d84-4d91-ba10-47ab007e61f4",
        "authorId" : "db47316c-0a85-441a-8891-3b84e8bfe474",
        "body" : "I guess I can put it in a new test file and run it under a harness that asserts the test fails and greps the message.",
        "createdAt" : "2020-01-31T00:12:14Z",
        "updatedAt" : "2020-01-31T00:55:42Z",
        "lastEditedBy" : "db47316c-0a85-441a-8891-3b84e8bfe474",
        "tags" : [
        ]
      },
      {
        "id" : "02e2c7a6-600c-4c1d-8032-e5498f4dea23",
        "parentId" : "97e8a10f-6d84-4d91-ba10-47ab007e61f4",
        "authorId" : "db47316c-0a85-441a-8891-3b84e8bfe474",
        "body" : "Ok - now it should only print the warning if a test fails and the HttpClient was created.",
        "createdAt" : "2020-01-31T00:38:06Z",
        "updatedAt" : "2020-01-31T00:55:42Z",
        "lastEditedBy" : "db47316c-0a85-441a-8891-3b84e8bfe474",
        "tags" : [
        ]
      }
    ],
    "commit" : "c0a373ae5fc6574647aae5babf59c02dd116df93",
    "line" : 17,
    "diffHunk" : "@@ -1,1 +80,84 @@  @override\n  HttpClient createHttpClient(SecurityContext _) {\n    if (!warningPrinted) {\n      test_package.printOnFailure(\n        'Warning: At least one test in this suite creates an HttpClient. When\\n'"
  }
]